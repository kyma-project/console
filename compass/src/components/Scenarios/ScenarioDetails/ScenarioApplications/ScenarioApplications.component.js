import React from 'react';
import PropTypes from 'prop-types';
import LuigiClient from '@kyma-project/luigi-client';

import { GenericList } from 'react-shared';
import AssignEntityToScenarioModal from './../shared/AssignEntityToScenarioModal/AssignApplicationsToScenarioModal.container';
import unassignScenarioHandler from './../shared/unassignScenarioHandler';

ScenarioApplications.propTypes = {
  scenarioName: PropTypes.string.isRequired,
  getApplicationsForScenario: PropTypes.object.isRequired,
  removeApplicationFromScenario: PropTypes.func.isRequired,
  sendNotification: PropTypes.func.isRequired,
};

export default function ScenarioApplications({
  scenarioName,
  getApplicationsForScenario,
  removeApplicationFromScenario,
  sendNotification,
}) {
  if (getApplicationsForScenario.loading) {
    return <p>Loading...</p>;
  }
  if (getApplicationsForScenario.error) {
    return `Error! ${getApplicationsForScenario.error.message}`;
  }

  const deleteHandler = async application => {
    const showSuccessNotification = applicationName => {
      sendNotification({
        variables: {
          content: `Unassigned "${applicationName}" from ${scenarioName}.`,
          title: applicationName,
          color: '#359c46',
          icon: 'accept',
          instanceName: scenarioName,
        },
      });
    };
    if (application.labels.scenarios.length === 1) {
      LuigiClient.uxManager().showAlert({
        text: 'At least one scenario is required.',
        type: 'warning',
        closeAfter: 10000,
      });
      return;
    }

    await unassignScenarioHandler(
      application.name,
      application.id,
      application.labels.scenarios,
      removeApplicationFromScenario,
      scenarioName,
      async () => {
        showSuccessNotification(application.name);
        await getApplicationsForScenario.refetch();
      },
    );
  };

  const headerRenderer = () => ['Name', 'Total APIs'];

  const rowRenderer = application => [
    application.name,
    application.apis.totalCount + application.eventAPIs.totalCount,
  ];

  const actions = [
    {
      name: 'Unassign',
      handler: deleteHandler,
    },
  ];

  const assignedApplications = getApplicationsForScenario.applications.data;

  const extraHeaderContent = (
    <AssignEntityToScenarioModal
      originalEntities={assignedApplications}
      getEntitiesForScenario={getApplicationsForScenario}
    />
  );

  return (
    <GenericList
      extraHeaderContent={extraHeaderContent}
      title="Applications"
      notFoundMessage="No Applications for this Scenario"
      entries={assignedApplications}
      headerRenderer={headerRenderer}
      actions={actions}
      rowRenderer={rowRenderer}
      showSearchField={false}
    />
  );
}
