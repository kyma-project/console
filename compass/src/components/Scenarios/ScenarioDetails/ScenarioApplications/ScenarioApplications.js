import React, { useContext } from 'react';
import LuigiClient from '@kyma-project/luigi-client';
import { useQuery, useMutation } from '@apollo/react-hooks';

import { GenericList } from 'react-shared';
import AssignEntityToScenarioModal from './../shared/AssignEntityToScenarioModal/AssignApplicationsToScenarioModal.container';
import unassignScenarioHandler from './../shared/unassignScenarioHandler';
import {
  createEqualityQuery,
  GET_APPLICATIONS_FOR_SCENARIO,
  SET_APPLICATION_SCENARIOS,
} from '../../gql';
import ScenarioNameContext from '../ScenarioNameContext';
import { SEND_NOTIFICATION } from '../../../../gql';

export default function ScenarioApplications({ updateApplicationsCount }) {
  const scenarioName = useContext(ScenarioNameContext);
  const [sendNotification] = useMutation(SEND_NOTIFICATION);

  let {
    data: applicationsForScenario,
    error,
    loading,
    refetch: refetchApplications,
  } = useQuery(GET_APPLICATIONS_FOR_SCENARIO, {
    variables: {
      filter: [
        {
          key: 'scenarios',
          query: createEqualityQuery(scenarioName),
        },
      ],
    },
  });

  const [removeApplicationFromScenario] = useMutation(
    SET_APPLICATION_SCENARIOS,
  );

  if (loading) {
    return <p>Loading...</p>;
  }
  if (error) {
    return <p>`Error! ${error.message}`;</p>;
  }

  updateApplicationsCount(applicationsForScenario.applications.totalCount);

  const deleteHandler = async application => {
    const showSuccessNotification = applicationName => {
      sendNotification({
        variables: {
          content: `Unassigned "${applicationName}" from ${scenarioName}.`,
          title: applicationName,
          color: '#359c46',
          icon: 'accept',
          instanceName: scenarioName,
        },
      });
    };
    if (application.labels.scenarios.length === 1) {
      LuigiClient.uxManager().showAlert({
        text: 'At least one scenario is required.',
        type: 'warning',
        closeAfter: 10000,
      });
      return;
    }

    await unassignScenarioHandler(
      application.name,
      application.id,
      application.labels.scenarios,
      removeApplicationFromScenario,
      undefined,
      scenarioName,
      async () => {
        await refetchApplications();
        updateApplicationsCount(
          applicationsForScenario.applications.totalCount,
        );
        showSuccessNotification(application.name);
      },
    );
  };

  const headerRenderer = () => ['Name', 'Total APIs'];

  const rowRenderer = application => [
    application.name,
    application.apiDefinitions.totalCount +
      application.eventDefinitions.totalCount,
  ];

  const actions = [
    {
      name: 'Unassign',
      handler: deleteHandler,
    },
  ];

  const assignedApplications = applicationsForScenario.applications.data;

  const extraHeaderContent = (
    <AssignEntityToScenarioModal
      originalEntities={assignedApplications}
      entitiesForScenarioRefetchFn={refetchApplications}
    />
  );

  return (
    <GenericList
      extraHeaderContent={extraHeaderContent}
      title="Applications"
      notFoundMessage="No Applications for this Scenario"
      entries={assignedApplications}
      headerRenderer={headerRenderer}
      actions={actions}
      rowRenderer={rowRenderer}
      showSearchField={false}
    />
  );
}
