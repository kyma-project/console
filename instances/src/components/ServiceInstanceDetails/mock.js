export const Mock = {
  __typename: 'clusterDocsTopic',
  name: 'kyma',
  groupName: 'root',
  displayName: 'Kyma',
  description: 'Overall documentation for Kyma',
  assets: [
    {
      __typename: 'clusterAsset',
      name: 'asset1',
      type: 'markdown',
      files: [
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/01-01-in-a-nutshell.md',
          metadata: {
            __typename: 'fileMetadata',
            title: 'In a nutshell',
            type: 'Overview',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/04-01-overview.md',
          metadata: {
            __typename: 'fileMetadata',
            title: 'Overview',
            type: 'Installation',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/04-03-cluster-installation.md',
          metadata: {
            __typename: 'fileMetadata',
            // title: 'Install Kyma on a cluster',
            type: 'Installation',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/12-01-try-out-kyma.md',
          metadata: {
            __typename: 'fileMetadata',
            type: 'Examples',
            title: 'Try out Kyma',
          },
        },
      ],
    },
  ],
};
