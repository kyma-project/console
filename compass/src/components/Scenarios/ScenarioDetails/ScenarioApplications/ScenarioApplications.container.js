import { graphql, compose } from 'react-apollo';
import { fromRenderProps } from 'recompose';
import {
  GET_APPLICATIONS_FOR_SCENARIO,
  SET_APPLICATION_SCENARIOS,
  createEqualityQuery,
} from '../../gql';
import { SEND_NOTIFICATION } from '../../../../gql';

import ScenarioApplications from './ScenarioApplications.component';
import ScenarioNameContext from './../ScenarioNameContext';

export default compose(
  fromRenderProps(ScenarioNameContext.Consumer, scenarioName => ({
    scenarioName,
  })),
  graphql(SET_APPLICATION_SCENARIOS, {
    props: ({ mutate }) => ({
      removeApplicationFromScenario: async (id, scenarios) =>
        await mutate({
          variables: {
            id,
            scenarios,
          },
        }),
    }),
  }),
  graphql(GET_APPLICATIONS_FOR_SCENARIO, {
    name: 'getApplicationsForScenario',
    options: ({ scenarioName }) => {
      const filter = {
        key: 'scenarios',
        query: createEqualityQuery(scenarioName),
      };
      return {
        variables: {
          filter: [filter],
        },
      };
    },
  }),
  graphql(SEND_NOTIFICATION, {
    name: 'sendNotification',
  }),
)(ScenarioApplications);
