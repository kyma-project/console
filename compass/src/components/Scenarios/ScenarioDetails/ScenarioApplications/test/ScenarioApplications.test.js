import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ScenarioApplications from '../ScenarioApplications.component';
import { responseMock } from './mock';
jest.mock('react-shared', () => ({
  GenericList: function GenericListMocked(props) {
    return 'generic-list-mocked-content';
  },
}));
describe('ScenarioApplications', () => {
  it('Renders with minimal props', () => {
    const component = shallow(
      <ScenarioApplications
        scenarioName="scenario name"
        getApplicationsForScenario={responseMock}
        removeApplicationFromScenario={() => {}}
        sendNotification={() => {}}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
