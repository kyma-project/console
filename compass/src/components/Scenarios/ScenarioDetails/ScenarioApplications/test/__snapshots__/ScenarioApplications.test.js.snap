// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScenarioApplications Renders with minimal props 1`] = `
<t
  actions={
    Array [
      Object {
        "handler": [Function],
        "name": "Unassign",
      },
    ]
  }
  entries={
    Array [
      Object {
        "apis": Object {
          "totalCount": 3,
        },
        "eventAPIs": Object {
          "totalCount": 5,
        },
        "id": "id 1",
        "labels": Object {},
        "name": "Application 1",
      },
    ]
  }
  extraHeaderContent={
    <Apollo(Apollo(Apollo(withProps(AssignEntityToScenarioModal))))
      getEntitiesForScenario={
        Object {
          "applications": Object {
            "data": Array [
              Object {
                "apis": Object {
                  "totalCount": 3,
                },
                "eventAPIs": Object {
                  "totalCount": 5,
                },
                "id": "id 1",
                "labels": Object {},
                "name": "Application 1",
              },
            ],
            "totalCount": 1,
          },
          "error": null,
          "loading": false,
        }
      }
      originalEntities={
        Array [
          Object {
            "apis": Object {
              "totalCount": 3,
            },
            "eventAPIs": Object {
              "totalCount": 5,
            },
            "id": "id 1",
            "labels": Object {},
            "name": "Application 1",
          },
        ]
      }
    />
  }
  headerRenderer={[Function]}
  notFoundMessage="No Applications for this Scenario"
  rowRenderer={[Function]}
  showSearchField={false}
  textSearchProperties={
    Array [
      "name",
      "description",
    ]
  }
  title="Applications"
/>
`;
